### PR TITLE
Fix LSTM benchmark to evaluate on test set

### LINKED ISSUE #262: Using train_set for evaluation in MackeyGlass LSTM benchmark?
Hello NeuroBench team,

I was reviewing the `lstm_benchmark.py` example for the Mackey-Glass dataset and noticed a potential issue regarding the evaluation setup. In the benchmark function, specifically at [line 168](https://github.com/NeuroBench/neurobench/blob/main/examples/mackey_glass/lstm_benchmark.py#L168), it appears the `train_set` is being used for evaluation:

```python
    test_set_loader = DataLoader(train_set,
                                 batch_size=mg.testtime_pts,
                                 shuffle=False)
```
In addition, I cannot make sense out of why batch size is set to `testtime_pts`.
The actual `test_set` defined above in [line 117](https://github.com/NeuroBench/neurobench/blob/main/examples/mackey_glass/lstm_benchmark.py#L117) is not used anywhere else in the benchmark script.

I have also changed it to use the actual `test_set` but then performance gets much worse with sMAPEs of around 30.

Is it just me not understanding the code or is there a potential issue? I'd appreciate if you could clarify the reason behind using `train_set` for this particular evaluation step in the benchmark, or if this might be an oversight?

### PR DESCRIPTION
**Pull request by YounesBouhadjar**
Fixes #262 
The benchmark is corrected to evaluate LSTM on the test set, using the hidden states warmed up at the end of training.

### CODE REVIEW

**Comment by jasonlyik** [comment]
Thanks @YounesBouhadjar, I think there was just one small issue -

When the hidden states are populated in the last epoch of the training at [L139](https://github.com/NeuroBench/neurobench/blob/correct_mg_bench/examples/mackey_glass/lstm_benchmark.py#L139), there is still an optimizer step / weight update after that. 

So I believe that the hidden states initialized for testing would be using the weights after the second-to-last update, rather than the weights after the final update.

I changed it to re-run the forward pass on the train data before the benchmark run on the test set, and also re-did the benchmark, but the difference in results is negligible.

**Review by jasonlyik — APPROVED** [review]
Update corrects the issue with the test dataset for the LSTM Mackey-Glass benchmark.
